### PR TITLE
#12021 Release 23.10.0

### LINKED ISSUE #12021: Release 23.10.0
This is the PR to track the next release of twisted

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #12021 

Follow the steps from https://docs.twisted.org/en/latest/development/release-process.html

### CODE REVIEW

**Comment by adiroiban** [comment]
This is ready for review  needs-review 

GitHub Release at https://github.com/twisted/twisted/releases/tag/twisted-23.10.0rc1

GitHub Actions job at https://github.com/twisted/twisted/actions/runs/6633431158/job/18021141469

PyPi at https://pypi.org/project/Twisted/23.10.0rc1/

Documentation at https://twisted--12022.org.readthedocs.build/en/12022/

**Comment by will-lx** [comment]
Looks like Python 3.8 / macOS 12 tests are [timing out](https://github.com/twisted/twisted/actions/runs/6637559051/job/18032120552?pr=12022#step:10:14763) 
```
twisted.internet.test.reactormixins.TestTimeoutError: reactor still running after 120.0 seconds
```
Is the reactor framework flaky? Might it simply need to be re-run in this case?

**Comment by glyph** [comment]
sadly our test matrix is large enough that intermittent failures in the github actions executions substrate routinely cause otherwise-fine PRs to fail for some unrelated reason. Sometimes this is a fault in Twisted's tests, but sometimes it's just luck of the draw with the free infrastructure we're hosting our tests on.

I've re-run the tests and the one failing builder seems to be passing now.

**Comment by adiroiban** [comment]
I don't think the flaky test is caused caused by GitHub infra.

I have create a separate ticket to track that test https://github.com/twisted/twisted/issues/12029

I spent some time trying to debug this without luck.

I think that is best to just disable that test on macOS for now, as it's a distraction.

If anyone cares about Twisted on macOS they can help with that ticket.

--------

@glyph can you please review this PR so that we can do the release?

thanks

**Comment by glyph** [comment]
> I don't think the flaky test is caused caused by GitHub infra.

I've been seeing a couple of "build didn't even start" failure modes recently, which is why I mentioned this, but yes, this looks like a real bug.

File: NEWS.rst [hunk 1]
```diff
@@ -8,6 +8,98 @@ https://twisted.org/trac/ticket/<number>
 
 .. towncrier release notes start
 
+Twisted 23.10.0.rc1 (2023-10-24)
+================================
+
+Features
+--------
+
+- twisted.python.filepath.FilePath and related classes (twisted.python.filepath.IFilepath, twisted.python.filepath.AbstractFilePath, twisted.python.zippath.ZipPath, and twisted.python.zippath.ZipArchive) now have type annotations.  Additionally, FilePath is now generic, describing its mode, so you can annotate variables as FilePath[str] or FilePath[bytes] depending on the types that you wish to get back from the 'path' attribute and related methods like 'basename'. (#11822)
```
Comment (glyph) [inline NEWS.rst:20]: ✨ very excited to get this released

**Review by glyph — APPROVED** [review]
nice, love to see it. sometimes it feels like Twisted isn't getting a ton of activity, but dang, look at all those news fragments :)

**Comment by adiroiban** [comment]
Thanks Glyph for the review.

The release notes is asking for another round of reviews after the last commit... but I think we are good.

I will trigger the release as soon as the tests are green and after that update to .dev0 and merge

**Comment by adiroiban** [comment]
Final release on GitHub https://github.com/twisted/twisted/releases/tag/twisted-23.10.0

GitHub Actions job https://github.com/twisted/twisted/actions/runs/6703716105/job/18214766790

PyPi at https://pypi.org/project/Twisted/23.10.0/

--------

I will push the dev0 and then merge

**Pull request merged**